precompiles/ArbGasInfo.go

```diff
@@ -18,14 +18,47 @@ type ArbGasInfo struct {
 	Address addr // 0x6c
 }
 
-var zero = big.NewInt(0)
 var storageArbGas = big.NewInt(int64(storage.StorageWriteCost))
 
 // Get prices in wei when using the provided aggregator
 func (con ArbGasInfo) GetPricesInWeiWithAggregator(
 	c ctx,
 	evm mech,
 	aggregator addr,
+) (huge, huge, huge, huge, huge, huge, error) {
+	if c.State.FormatVersion() < 4 {
+		return con._preVersion4_GetPricesInWeiWithAggregator(c, evm, aggregator)
+	}
+
+	l1GasPrice, err := c.State.L1PricingState().PricePerUnit()
+	if err != nil {
+		return nil, nil, nil, nil, nil, nil, err
+	}
+	l2GasPrice := evm.Context.BaseFee
+
+	// aggregators compress calldata, so we must estimate accordingly
+	weiForL1Calldata := arbmath.BigMulByUint(l1GasPrice, params.TxDataNonZeroGasEIP2028)
+
+	// the cost of a simple tx without calldata
+	perL2Tx := arbmath.BigMulByUint(weiForL1Calldata, l1pricing.TxFixedCostEstimate)
+
+	// nitro's compute-centric l2 gas pricing has no special compute component that rises independently
+	perArbGasBase, err := c.State.L2PricingState().MinBaseFeeWei()
+	if err != nil {
+		return nil, nil, nil, nil, nil, nil, err
+	}
+	perArbGasCongestion := arbmath.BigSub(l2GasPrice, perArbGasBase)
```

Comment:
This could be negative during gas estimation / eth calls

Reply:
Fixed.